code/modules/games/cards/playing_cards.dm

```diff
@@ -24,6 +24,10 @@
 		hand.currenthand.Remove(card)
 		user.put_in_hands(card)
 		hand.update_icon()
+		if(hand.currenthand.len == 1)
+			var/obj/item/toy/singlecard/C = hand.currenthand[1]
+			qdel(hand)
```

Comment:
Move the qdel after the put_in_inactive_hand() to avoid potential race condition issues with the card getting nuked before it can be placed in the hand.